### PR TITLE
Add support for continuation lines in `Headers::try_from`

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: src/headers.rs [hunk 1]
```diff
@@ -157,12 +161,19 @@ impl TryFrom<&[u8]> for Headers {
...
         };
 
-        for line in lines {
+        while let Some(line) = lines.next() {
             let splits = line.splitn(2, ':').map(str::trim).collect::<Vec<_>>();
             match splits[..] {
                 [k, v] => {
                     let entry = inner.entry(k.to_string()).or_insert_with(HashSet::default);
-                    entry.insert(v.to_string());
+
```
Comment (derekcollison) [inline src/headers.rs:169]: Does this skip efficiently when its just **NATS/1.0**

Comment (caspervonb) [reply src/headers.rs:169]: If the entire header is just `NATS/1.0` it wouldn't reach inside the loop at all as `next()` above would return `None` on the first call.

Comment (derekcollison) [reply src/headers.rs:169]: nvm this just cleans up the normal k/v insertion code, got it.

**Review by derekcollison — APPROVED** [review]
LGTM

File: src/headers.rs (hunk 1 above)
Comment (caspervonb) [reply src/headers.rs:169]: Sort of, It's turned into a "peekable" iter to allow continuation lines to "take" additional lines below.

**Pull request merged**